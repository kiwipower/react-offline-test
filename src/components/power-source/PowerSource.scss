.PowerSource {
  margin: 30px;
  display: flex;
  flex-direction: column;
  align-items: center;

  @mixin power-source-size {
    width: 64px;
    height: auto;
  }

  .other {
    @include power-source-size;
    content:url('../../assets/images/power-source-icons/factory.png');
  }

  .hydro {
    @include power-source-size;
    content:url('../../assets/images/power-source-icons/water-drop.png');
  }

  .imports {
    @include power-source-size;
    content:url('../../assets/images/power-source-icons/networking.png');
  }

  .coal {
    @include power-source-size;
    content:url('../../assets/images/power-source-icons/wagon.png');
  }

  .nuclear {
    @include power-source-size;
    content:url('../../assets/images/power-source-icons/nuclear-plant.png');
  }

  .biomass {
    @include power-source-size;
    content:url('../../assets/images/power-source-icons/trash.png');
  }

  .wind {
    @include power-source-size;
    content:url('../../assets/images/power-source-icons/eolic-energy.png');
  }

  .solar {
    @include power-source-size;
    content:url('../../assets/images/power-source-icons/solar-panel.png');
  }

  .gas {
    @include power-source-size;
    content:url('../../assets/images/power-source-icons/gas.png');
  }

  .title {
    color: #4B0082;
    font-size: 18px;
    font-weight: bold;
  }

  .percent {
    font-size: 36px;
    color: #FF7F00;
  }
}
