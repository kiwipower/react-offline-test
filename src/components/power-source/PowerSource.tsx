import * as React from 'react';

import './PowerSource.scss';

interface Props {
    fuel: string,
    perc: number
}

export const PowerSource: React.FC<Props> =
    ({ fuel, perc }) => (
        <main className="PowerSource">
            <div><img className={fuel} alt={fuel}/></div>
            <div className="percent">{perc}</div>
            <div className="title">{fuel}</div>
        </main>
    );
