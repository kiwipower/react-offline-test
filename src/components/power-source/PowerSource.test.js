import React from 'react';
import { shallow } from 'enzyme';

import { PowerSource } from './PowerSource';

const defaultProps = {fuel: 'biomass', perc: 55.9};

describe('PowerSource', () => {
  test('should render correctly', () => {
    const component = shallow(<PowerSource {...defaultProps}/>);
    expect(component).toMatchSnapshot();
  });

  test('Should have a \\"PowerSource\\" class', () => {
    const wrapper = shallow(<PowerSource {...defaultProps}/>);
    expect(wrapper.hasClass('PowerSource')).toBe(true);
  });
});
