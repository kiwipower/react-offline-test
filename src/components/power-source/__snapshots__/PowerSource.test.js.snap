// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PowerSource should render correctly 1`] = `
<main
  className="PowerSource"
>
  <div>
    <img
      alt="biomass"
      className="biomass"
    />
  </div>
  <div
    className="percent"
  >
    55.9
  </div>
  <div
    className="title"
  >
    biomass
  </div>
</main>
`;
